src/main/java/roomescape/exception/ReservationExceptionHandler.java

```diff
@@ -0,0 +1,21 @@
+package roomescape.exception;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.util.NoSuchElementException;
+
+@RestControllerAdvice
+public class ReservationExceptionHandler {
+
+    @ExceptionHandler
+    protected ResponseEntity<ErrorResponse> handleIllegalArgumentException(IllegalArgumentException e) {
+        return ResponseEntity.badRequest().body(new ErrorResponse(e.getMessage()));
+    }
+
+    @ExceptionHandler
+    protected ResponseEntity<ErrorResponse> handleNoSuchElementException(NoSuchElementException e) {
+        return ResponseEntity.badRequest().body(new ErrorResponse(e.getMessage()));
+    }
```

Comment:
1. 접근 제어자가 protected 인 이유는?
2. 아래처럼 묶어서 사용해도 되지 않나요?

```suggestion
    @ExceptionHandler({IllegalArgumentException.class, NoSuchElementException.class})
    protected ResponseEntity<ErrorResponse> handleIllegalArgumentException(RuntimeException e) {
        return ResponseEntity.badRequest().body(new ErrorResponse(e.getMessage()));
    }
```